packages/cozy-stack-client/src/SharingCollection.js

```diff
@@ -123,31 +123,54 @@ class SharingCollection extends DocumentCollection {
    * Add an array of contacts to the Sharing
    *
    * @param {object} sharing Sharing Object
-   * @param {Array} recipients Array of {id:1, type:"io.cozy.contacts"}
-   * @param {string} sharingType Read and write: two-way. Other only read
+   * @param {Array=} recipients Array of {id:1, type:"io.cozy.contacts"}
+   * @param {string=} sharingType Read and write: two-way. Other only read
    */
   async addRecipients(sharing, recipients, sharingType) {
-    const resp = await this.stackClient.fetchJSON(
-      'POST',
-      uri`/sharings/${sharing._id}/recipients`,
-      {
-        data: {
-          type: 'io.cozy.sharings',
-          id: sharing._id,
-          relationships:
-            sharingType === 'two-way'
-              ? {
-                  recipients: { data: recipients.map(toRelationshipItem) }
-                }
-              : {
-                  read_only_recipients: {
-                    data: recipients.map(toRelationshipItem)
+    if (recipients !== undefined) {
+      console.warn(`addRecipients(sharing, recipients, sharingType is deprecated) use : \n 
+      addRecipients({document, recipients,readOnlyRecipients})`)
+      const resp = await this.stackClient.fetchJSON(
+        'POST',
+        uri`/sharings/${sharing._id}/recipients`,
+        {
+          data: {
+            type: 'io.cozy.sharings',
+            id: sharing._id,
+            relationships:
+              sharingType === 'two-way'
+                ? {
+                    recipients: { data: recipients.map(toRelationshipItem) }
+                  }
+                : {
+                    read_only_recipients: {
+                      data: recipients.map(toRelationshipItem)
+                    }
                   }
-                }
+          }
```

Comment:
Nit: I think in those cases, moving the old code to addRecipientsDeprecated would make the commit a bit more readable as the indentation would not change.

Reply:
You know you're not stupid!  <3

Reply:
Removed the old API and created a BC